libbeat/beat/beat.go

```diff
@@ -193,5 +211,22 @@ func (b *Beat) Run() {
 
 // Stop calls the beater Stop action.
 func (beat *Beat) Stop() {
+	logp.Info("Stopping Beat")
 	beat.BT.Stop()
+	close(beat.done)
+}
+
+// Reload the beat for example with a fresh configuration
+// This requires that stop and cleanup properly shuts down the beat
```

Comment:
It would be nice to have something like a "counter" on how many go routines are still running. Not sure if something like this exists? If we have such a counter, we could check if too many goroutines are running and output an error message.

As Reload is currently not used, I can also remove it from this PR so we can delay the discussion here.